src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,67 @@
+package calculator
+
+import java.util.regex.Pattern
+
+const val CUSTOM_SPLITTER_LOCATION = 2
+private const val CUSTOM_SPLITTER = "//(.)\\n(.*)"
+private val DEFAULT_SPLITTER = listOf(",", ":")
+const val CUSTOM_SPLITTER_LAST_LOCATION = 4
+
+class Calculator(private val numbersInput: String) {
+    private var isCustomSplitter: Boolean = false
+    private var splitters: List<String> = DEFAULT_SPLITTER
+    fun execute(): Int {
+        getNumbersForCalculator(numbersInput)
+        hasCustomSplitter()
+        setSplitters()
+        hasOnlyValidString()
+        val numbers = parsing()
+        return calculate(numbers)
+    }
+
+    private fun calculate(numbers: List<Number>): Int {
+        return numbers.sumBy { it.toInt() }
+    }
+
+    private fun getNumbersForCalculator(numbersInput: String): String {
+        require(!numbersInput.isBlank()) { throw CalculatorException("숫자를 입력해주세요.") }
+        return numbersInput
+    }
+
+    private fun hasCustomSplitter() {
+        try {
+            isCustomSplitter = Pattern.compile(CUSTOM_SPLITTER).matcher(numbersInput).matches()
+        } catch (e: CalculatorException) {
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        } catch (e: Exception) {
```